Dockerfile

```
@@ -28,33 +36,64 @@ LABEL summary="$SUMMARY" \
     com.redhat.component="python38-docker" \
     name="Koku" \
     version="1" \
-    maintainer="Red Hat Cost Management Services"
+    maintainer="Red Hat Cost Management Services <cost-mgmt@redhat.com>"
 
-USER root
+# Very minimal set of packages
+# glibc-langpack-en is needed to set locale to en_US and disable warning about it
+# gcc to compile some python packages (e.g. ciso8601)
+# shadow-utils to make useradd available
+RUN INSTALL_PKGS="python38 python38-devel glibc-langpack-en gcc shadow-utils" && \
+    microdnf --nodocs -y upgrade && \
+    microdnf -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 install $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    microdnf -y clean all --enablerepo='*'
 
-# Get latest packages
-# RUN dnf -y upgrade
+WORKDIR ${HOME}
 
-COPY ./.s2i/bin/ $STI_SCRIPTS_PATH
+# Create a Python virtual environment for use by any application to avoid
+# potential conflicts with Python packages preinstalled in the main Python
+# installation.
+RUN python3.8 -m venv /pipenv-venv
+ENV PATH="/pipenv-venv/bin:$PATH"
+# Install pipenv into the virtual env
+RUN \
+    pip install --upgrade pip && \
+    pip install pipenv
 
-# Copy application files to the image.
-COPY . /tmp/src/.
+# install dependencies
+COPY Pipfile .
+COPY Pipfile.lock .
+RUN \
+    # install the dependencies into the working dir (i.e. /koku/.venv)
+    pipenv install --deploy && \
+    # delete the pipenv cache
+    pipenv --clear
 
+# Runtime env variables:
+ENV \
+    # Add the koku virtual env bin to the front of PATH.
+    # This activates the virtual env for all subsequent python calls.
+    PATH="$VIRTUAL_ENV_DIR/bin:$PATH" \
+    PROMETHEUS_MULTIPROC_DIR=/tmp
 
-RUN \
-    /usr/bin/fix-permissions /tmp/src && \
-    chmod 755 $STI_SCRIPTS_PATH/assemble $STI_SCRIPTS_PATH/run && \
-    groupadd -g ${USER_ID} koku && \
-    useradd -m -s /bin/bash -g ${USER_ID} -u ${USER_ID} -G root koku && \
-    chmod g+rwx /opt
+# copy the src files into the workdir
+COPY . .
 
+# create the koku user
+RUN \
+    adduser koku -u ${USER_ID} -g 0 && \
+    chmod ug+rw ${HOME} ${HOME}/koku ${HOME}/koku/static /tmp
 USER koku
 
-RUN umask u=rwx,g=rwx,o=rx
+# create the static files
+RUN python koku/manage.py collectstatic --noinput
```

If there's a chance that an existing `app.log` will be copied into the image via line 80, I suggest this:
```suggestion
RUN rm -f ${HOME}/koku/app.log && python koku/manage.py collectstatic --noinput && rm -f ${HOME}/koku/app.log
```

---

I'll add a .dockerignore entry for `.log` files.